boa/src/builtins/function/mod.rs

```diff
@@ -329,6 +329,79 @@ impl BuiltInFunctionObject {
         // TODO?: 5. PrepareForTailCall
         context.call(this, &this_arg, &arg_list)
     }
+
+    fn to_string(this: &Value, _: &[Value], context: &mut Context) -> Result<Value> {
+        let name = {
+            // Is there a case here where if there is no name field on a value
+            // We it should default to None? Do all functions have names set?
+            let value = this.get_field("name", &mut *context)?;
+            if value.is_null_or_undefined() {
+                None
+            } else {
+                Some(value.to_string(context)?)
+            }
+        };
+
+        let function = {
+            let object = this.as_object().map(|object| {
+                let func = object.borrow().as_function().map(|f| f.clone());
```

Comment:
Here, you are using `|f| f.clone()` as a closure, when you could use `as_function().cloned()`. This is causing a `clippy` error.